core/src/main/scala/org/apache/spark/package.scala

```diff
@@ -41,7 +41,58 @@ package org.apache
  * level interfaces. These are subject to changes or removal in minor releases.
  */
 
+import java.util.Properties
+
 package object spark {
-  // For package docs only
-  val SPARK_VERSION = "2.0.0-preview"
+
+  private object SparkBuildInfo {
```

Comment:
@vanzin 

With the current state being in the preview mode, users testing the framework would see the version as 2.0.0-SNAPSHOT as being mentioned in the pom file.

If this is acceptable, rest of the changes are as is.